smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestRequestGenerator.kt

```diff
@@ -68,8 +84,26 @@ open class HttpProtocolUnitTestRequestGenerator protected constructor(builder: B
         }
     }
 
+    private fun renderClientBlock(test: HttpRequestTestCase) {
+        val serviceShape = ctx.service
+        val clientName = "${ctx.settings.sdkId}Client"
+
+        if (!serviceShape.getTrait(EndpointRuleSetTrait::class.java).isPresent) {
+            val host: String? = test.host.orElse(null)
+            val url: String = "http://${host ?: "example.com"}"
+            writer.write("\nlet config = try await ${clientName}.${clientName}Configuration(endpointResolver: StaticEndpointResolver(endpoint: try \$N(urlString: \$S)))", SmithyHTTPAPITypes.Endpoint, url)
+        } else {
+            writer.write("\nlet config = try await ${clientName}.${clientName}Configuration()")
+        }
+        writer.write("config.region = \"us-west-2\"")
+        writer.write("config.httpClientEngine = ProtocolTestClient()")
+        writer.write("config.idempotencyTokenGenerator = ProtocolTestIdempotencyTokenGenerator()")
```

Comment:
Use our own token generator to ensure the expected test token will get generated during middleware execution